This patch is necessary to allow the jsonapi_include module to recognize the
jsonapi upload route as a jsonapi request. Then the response will be parsed
correctly.

diff --git a/core/modules/jsonapi/src/Routing/Routes.php b/core/modules/jsonapi/src/Routing/Routes.php
index 609ed91..4b3cf00 100644
--- a/core/modules/jsonapi/src/Routing/Routes.php
+++ b/core/modules/jsonapi/src/Routing/Routes.php
@@ -265,7 +265,8 @@ protected static function getFileUploadRoutesForResourceType(ResourceType $resou
    */
   public static function isJsonApiRequest(array $defaults) {
     return isset($defaults[RouteObjectInterface::CONTROLLER_NAME])
-      && str_starts_with($defaults[RouteObjectInterface::CONTROLLER_NAME], static::CONTROLLER_SERVICE_NAME);
+      && (str_starts_with($defaults[RouteObjectInterface::CONTROLLER_NAME], static::CONTROLLER_SERVICE_NAME)
+      ||  str_starts_with($defaults[RouteObjectInterface::CONTROLLER_NAME], 'jsonapi.file_upload'));
   }

   /**
